File: four_pillars_of_oop/no_01_encapsulation/no_04_security/python/UserAccount.py
class UserAccount:
    def __init__(self, username, password):
        self.__username = username
        self.__password = self.__encrypt_password(password)  # Private attribute

    def __encrypt_password(self, password):
        # Simplified encryption (not for real use)
        return password[::-1]

    def check_password(self, password):
        return self.__password == self.__encrypt_password(password)

File: four_pillars_of_oop/no_01_encapsulation/no_04_security/java/UserAccount.java
package four_pillars_of_oop.no_01_encapsulation.no_04_security.java;

public class UserAccount {
    private String username;  // Private attributes
    private String encryptedPassword;

    public UserAccount(String username, String password) {
        this.username = username;
        this.encryptedPassword = encryptPassword(password);
    }

    private String encryptPassword(String password) {
        // Simplified encryption (for demonstration purposes)
        return new StringBuilder(password).reverse().toString();
    }

    public boolean checkPassword(String password) {
        return encryptedPassword.equals(encryptPassword(password));
    }
}
